### PR TITLE
test: use the jaeger installed on svc01 for CI tests 

### PR DESCRIPTION
**Pull request by tiagolobocastro**
feat: add jaegertracing processor tags

Each component can now receive a list of tags via cmdline or env variable.
The deployer is enabled to do the same and can also add them directly to jaeger.

---------
test: add BUILD_TAG and STAGE_NAME to deployer

Add the given contents of the tags as run and run.stage, which the deployer
then sets as jaeger processor tags which get set on all spans that pass
through the jaeger collector.

---------
feat(deployer): add support for an external jaeger collector

Adds a new cmdline argument or env variable that tells the deployer to set the
local jaeger instance an agent that sends all the spans to an external collector.

---------
To enable an external jaeger on CI, simply add the env variable `EXTERNAL_JAEGER` and set it to the external jaeger collector

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
I've deleted the target dir, retrying...
bors try

File: composer/src/lib.rs [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ impl ComposeTest {
                         );
                     }
                     if !port_map.is_empty() {
-                        host_config.port_bindings = Some(port_map);
+                        //host_config.port_bindings = Some(port_map);
```
Comment (paulyoong) [inline composer/src/lib.rs:1175]: Should this be removed?

File: control-plane/agents/core/src/server.rs [hunk 2]
```diff
@@ -10,9 +10,25 @@ use common::Service;
...
+// impl std::str::FromStr for KeyValArg {
+//     type Err = String;
+//
+//     fn from_str(source: &str) -> Result<Self, Self::Err> {
+//         let parts = source.split(':').collect::<Vec<_>>();
+//         if parts.len() == 2 {
+//             Ok()
+//         }
+//     }
+// }
```
Comment (paulyoong) [inline control-plane/agents/core/src/server.rs:31]: Remove this?

File: control-plane/agents/core/src/server.rs [hunk 3]
```diff
@@ -66,6 +86,28 @@ pub(crate) struct CliArgs {
...
+fn default_tracing_tags() -> Vec<KeyValue> {
+    use git_version::git_version;
+    vec![
+        KeyValue::new(
+            "git.commit",
+            git_version!(args = ["--abbrev=12", "--always"]),
+        ),
+        KeyValue::new("crate.version", env!("CARGO_PKG_VERSION")),
+    ]
+}
```
Comment (paulyoong) [inline control-plane/agents/core/src/server.rs:110]: This function seems to be duplicated in a number of packages. Can it be made common to all packages?
Similarly for `parse_key_value` and `default_tracing_tags`.

**Review by paulyoong — APPROVED** [review]

File: control-plane/agents/core/src/server.rs (hunk 3 above)
Comment (tiagolobocastro) [reply control-plane/agents/core/src/server.rs:110]: I've done it for the parse_key_value but we can't do it for default_tracing_tags as we don't want to share the values? I'm not sure actually

Comment (tiagolobocastro) [reply control-plane/agents/core/src/server.rs:88]: Done for both now

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors try

**Pull request merged**